### PR TITLE
Rework demos

### PR DESCRIPTION
**Pull request by pablogs9**
Related:
- https://github.com/micro-ROS/rclc/pull/58
- https://github.com/micro-ROS/rclc/pull/44

### CODE REVIEW

File: apps/add_two_ints_service/src/main.c [hunk 1]
```diff
@@ -10,55 +10,61 @@
...
-			example_interfaces__srv__AddTwoInts_Request req;
-			example_interfaces__srv__AddTwoInts_Request__init(&req);
-			RCSOFTCHECK(rcl_take_request(&serv,&req_id,&req))
+    // create executor
+    RCCHECK(rclc_executor_init(&executor, &support.context, 1, &allocator));
 
-			example_interfaces__srv__AddTwoInts_Response res;
-			example_interfaces__srv__AddTwoInts_Response__init(&res);
+    unsigned int rcl_wait_timeout = 10;   // in ms
+    RCCHECK(rclc_executor_set_timeout(&executor, RCL_MS_TO_NS(rcl_wait_timeout)));
```
Comment (JanStaschulat) [inline apps/add_two_ints_service/src/main.c]: not necessary to set timeout. The default timeout is 1s.

File: apps/add_two_ints_service/src/main.c [hunk 2]
```diff
@@ -10,55 +10,61 @@
...
+rcl_node_t node;
+rclc_support_t support;
+rcl_allocator_t allocator;
+rclc_executor_t executor;
+
+rcl_service_options_t service_options;
+rcl_service_t service;
+const rosidl_service_type_support_t * service_type_support;
+rcl_wait_set_t wait_set;
+rmw_request_id_t req_id;
```
Comment (JanStaschulat) [inline apps/add_two_ints_service/src/main.c]: why you define it as global variables?

I guess when you use rclc_ functions, you dont need wait_set, service_options, node_ops any more. Have a look what other vars are not necessary any more.

File: apps/add_two_ints_service/src/main.c [hunk 3]
```diff
@@ -10,55 +10,61 @@
...
+example_interfaces__srv__AddTwoInts_Response res;
+example_interfaces__srv__AddTwoInts_Request req;
+
```
Comment (JanStaschulat) [inline apps/add_two_ints_service/src/main.c]: also not necessary to have them as global vars, the service_callback gets res and req as parameters.

File: apps/add_two_ints_service/src/main.c (hunk 1 above)
Comment (pablogs9) [reply apps/add_two_ints_service/src/main.c]: One question, if I provide an executor timeout greater than the timeout value set in `rclc_executor_spin_some` which one will be the spin_some timeout?

Comment (JanStaschulat) [reply apps/add_two_ints_service/src/main.c]: The timeout parameter in `rclc_executor_spin_some()` is directly used as timeout for `rcl_wait()`. In this case, the default value or the value assigned with `rclc_executor_set_timeout` is not used.

When `rclc_executor_spin_some` is called from `rclc_executor_spin()`, then the default value ( or the value defined with `rclc_executor_set_timeout`, if the user calls this function) is used. 

See for example here: https://github.com/micro-ROS/rclc/blob/ad5da630ab4900e2b997e02be1ca0d81bebf70c1/rclc/src/rclc/executor.c#L1000

Comment (JanStaschulat) [reply apps/add_two_ints_service/src/main.c:62]: When it is called from executor_spin() then the default value (1s) or the value defined by the user is used. If you call the method directly, then you set the timeout: 
See for example here: https://github.com/micro-ROS/rclc/blob/ad5da630ab4900e2b997e02be1ca0d81bebf70c1/rclc/src/rclc/executor.c#L1000

File: apps/add_two_ints_service/src/main.c (hunk 2 above)
Comment (pablogs9) [reply apps/add_two_ints_service/src/main.c]: Reworked

File: apps/add_two_ints_service/src/main.c (hunk 3 above)
Comment (pablogs9) [reply apps/add_two_ints_service/src/main.c]: Reworked

File: apps/add_two_ints_service/src/main.c (hunk 1 above)
Comment (pablogs9) [reply apps/add_two_ints_service/src/main.c]: Thanks, I understand it now. I'm going to remove those lines.

**Review by jamoralp — APPROVED** [review]

**Pull request merged**